indy_node/server/request_handlers/domain_req_handlers/context_handler.py

```
@@ -0,0 +1,134 @@
+                                              [AuthActionAdd(txn_type=SET_CONTEXT,
+                                                             field='*',
+                                                             value='*')])
+
+    def gen_txn_id(self, txn):
+        self._validate_txn_type(txn)
+        path = ContextHandler.prepare_context_for_state(txn, path_only=True)
+        return path.decode()
+
+    def update_state(self, txn, prev_result, request, is_committed=False) -> None:
+        self._validate_txn_type(txn)
+        path, value_bytes = ContextHandler.prepare_context_for_state(txn)
+        self.state.set(path, value_bytes)
+
+    @staticmethod
+    def make_state_path_for_context(authors_did, context_name, context_version) -> bytes:
+        return "{DID}:{MARKER}:{CONTEXT_NAME}:{CONTEXT_VERSION}" \
+            .format(DID=authors_did,
+                    MARKER=MARKER_CONTEXT,
+                    CONTEXT_NAME=context_name,
+                    CONTEXT_VERSION=context_version).encode()
+
+    @staticmethod
+    def prepare_context_for_state(txn, path_only=False):
+        origin = get_from(txn)
+        context_name = get_txn_context_name(txn)
+        context_version = get_txn_context_version(txn)
+        value = {
+            META: get_txn_context_meta(txn),
+            DATA: get_txn_context_data(txn)
+        }
+        path = ContextHandler.make_state_path_for_context(origin, context_name, context_version)
+        if path_only:
+            return path
+        seq_no = get_seq_no(txn)
+        txn_time = get_txn_time(txn)
+        value_bytes = encode_state_value(value, seq_no, txn_time)
+        return path, value_bytes
+
+
+def _bad_uri(uri_string):
+    url = findall(URI_REGEX, uri_string)
+    if not url:
+        return True
+    return False
+
+
+def validate_data(data):
+    if isinstance(data, dict):
+        if "@context" not in data.keys():
+            raise Exception("data missing '@context' property")
```

Please raise more specific exceptions (`InvalidClientRequest` for example)

---

fixed in [3190909](https://github.com/hyperledger/indy-node/pull/1430/commits/31909099fb629c02c2ec62d4ade5d22ea723b8d3)